### DISCUSSION TITLE
Allow Strongly typed in multicall/readContracts with dynamic contracts parameter

### DISCUSSION THREAD

**Discussion by ChrisChan888** [wevm/wagmi#1888] [Q&A]
When I use multicall/readContracts, the contracts parameter is always based on a list of addresses, and
I will loop the list of addresses to construct the contracts parameter.
However, when I'm using a loop to create a contracts parameter.
The functionNames are not strongly typed.

There is no option hint in the image below:
![Screenshot 2023-02-20 at 2 31 49 PM](https://user-images.githubusercontent.com/86586482/220032018-c602f0d4-5ac7-4a50-b280-352a010da816.png)

I expect there is an option hint like the image below:
![Screenshot 2023-02-20 at 2 30 55 PM](https://user-images.githubusercontent.com/86586482/220032167-26371249-9826-4efc-a1b6-fe0209246aa4.png)

I try to do something like the image below, but there is no option hint.
![Screenshot 2023-02-20 at 2 38 26 PM](https://user-images.githubusercontent.com/86586482/220032382-db27117a-76f9-400f-81f2-6e34b2a6c9f1.png)

Any solution to make it works? For loop to create a list of contracts with strongly typed?

**Comment by azevedo** [comment]
https://github.com/wagmi-dev/wagmi/discussions/1897#discussioncomment-5490672 this worked for me to get `data` properly typed.
It doesn't help with intellisense though.
